nymms/api/routes.py

```diff
@@ -0,0 +1,142 @@
+import arrow
+
+from flask import request
+from flask.ext.api import FlaskAPI
+from flask.ext.api import status
+
+from schematics.exceptions import ValidationError
+
+from nymms.state import sdb_state
+from nymms.suppress import sdb_suppress
+from nymms.schemas import APIResult, APISuppression, APIStateRecord
+from nymms.config import config
+from nymms.utils import aws_helper
+from nymms import schemas
+
+
+nymms_api = FlaskAPI(__name__)
+
+
+@nymms_api.route('/state', methods=['GET'])
+def state():
+    """
+    List current states.
+    """
+    region = config.settings['region']
+    domain = config.settings['state_domain']
+
+    state = sdb_state.SDBStateBackend(region, domain)
+    states = state.get_all_states(
```

Comment:
Interesting.  I actually didn't know about the restrictions.  TBH, this was a pretty fast and furious effort, so there are certainly some issues and edge cases not worked out.  I have some other changes on my local branch to fix some other issues around filtering results.  I think just exposing the next token isn't a bad idea, so that the API user just adds that to their filter (request params).  I'll do a little experiment.

In general though I would say the if the response to /results has a non-null value for "next-token" (not currently even exposed), then the API user can just query /results?next_token=<value> to get the next batch, and continue.